pkg/operator/controller/machineconfig/controller.go

```diff
@@ -204,50 +220,54 @@ func (r *MachineConfigReconciler) inspectProfilingMCReq(ctx context.Context) err
 }
 
 // ensureProfConfEnabled is for enabling the profiling of requested services
-func (r *MachineConfigReconciler) ensureProfConfEnabled(ctx context.Context) (err error) {
+func (r *MachineConfigReconciler) ensureProfConfEnabled(ctx context.Context) (bool, error) {
 
 	var modCount, setEnabledCondition int
+	var err error
 	if modCount, err = r.ensureReqNodeLabelExists(ctx); err != nil {
 		r.Log.Error(err, "failed to ensure nodes are labelled")
 		// fails for even one node revert changes made
-		return r.revertNodeLabeling(ctx)
+		return true, r.revertNodeLabeling(ctx)
 	}
 	setEnabledCondition += modCount
 	if modCount, err = r.ensureReqMCPExists(ctx); err != nil {
 		r.Log.Error(err, "failed to ensure mcp exists")
-		return
+		return false, err
 	}
 	setEnabledCondition += modCount
```

Comment:
Sorry, question: 
I'm having trouble understanding why we  sum the number of labeled nodes with a +1 of MCP applied, MC applied.
If the MCP or MC are not applied we'd probably get an error and exit.
But if the number of labeled nodes isn't correct, wouldn't it go unnoticed? since we're just checking if `setEnabledCondition` > 0

Reply:
Used int type instead of bool to be aware of any new changes made and need to defer the MCP check to next reconciling call. If labeling a node fails, error is returned and it's retried in the next reconciliation.